File: java/src/main/java/com/nwmqpa/graphcall/nodes/Edge.java
package com.nwmqpa.graphcall.nodes;

import com.nwmqpa.graphcall.interfaces.ICompilable;

public class Edge implements ICompilable {

    private final String name;
    private String nesting;

    Edge(String name, String nesting) {
        this.name = name;
        this.nesting = nesting;
    }

    @Override
    public String compile() {
        StringBuilder sb = new StringBuilder();
        return sb.append(nesting).append(name).append("\n").toString();
    }
}

File: python/graphcall/nodes/edge.py
"""Module defining the edge of a Node."""
from graphcall.interfaces.compilable import Compilable


class Edge(Compilable):
    """Class defining the edge."""

    def __init__(self, name: str, nesting: str) -> None:
        """Initialize the object."""
        self.name = name
        self.nesting = nesting

    def compile(self):
        """Compile the edge."""
        return self.nesting + self.name + "\n"
